### PR TITLE
Support setCharacterEncoding with Charset

### LINKED ISSUE #26492: Servlet 6.1: Add overloaded setCharacterEncoding() methods that support Charset.
`https://github.com/jakartaee/servlet/issues/415`

### PR DESCRIPTION
**Pull request by pmd1nh**
fixes https://github.com/OpenLiberty/open-liberty/issues/26492

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/CharSetEncodingTest.war/src/charset/servlets/TestCharSetEncodingServlet.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        testRequestSetCharEncodingviaCharset(request, sBuilder);
+        testResponseSetCharEncodingviaCharset(response, sBuilder);
+        testServletContextSetCharset(context, sBuilder);
+
+        sos.println("Test Results from " + CLASS_NAME + ": \n" + sBuilder.toString());
+    }
+
+    private void testRequestSetCharEncodingviaCharset(HttpServletRequest request, StringBuilder sBuilder) {
+    	LOG(">>> Testing testRequestSetCharEncodingviaCharset");
+
```
Comment (pmd1nh) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/CharSetEncodingTest.war/src/charset/servlets/TestCharSetEncodingServlet.java]: Cleanup extra spaces

**Comment by pmd1nh** [comment]
#build
       #spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat,io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.6.0_fat_toleration.lWAS,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/bnd.bnd [hunk 2]
```diff
@@ -21,3 +23,6 @@ javac.target: 21
 fat.minimum.java.level: 21
 
 -buildpath: \
+	io.openliberty.jakarta.servlet.6.1;version=latest,\
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/bnd.bnd]: Can you fix the formatting here

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/CharSetEncodingTest.jar/src/sci/servlets/ServletContainerInitializerImpl.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private static final Logger LOG = Logger.getLogger(CLASS_NAME);
+
+    @Override
+    public void onStartup(Set<Class<?>> setOfClassesInterestedIn, ServletContext context) throws ServletException {
+        LOG.info("ServletContainerInitializerImpl.onStartup ENTER");
+        context.setRequestCharacterEncoding(Charset.defaultCharset());
+        context.setResponseCharacterEncoding(StandardCharsets.US_ASCII);
+        LOG.info("ServletContainerInitializerImpl.onStartup RETURN");
+    }
+}
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/CharSetEncodingTest.jar/src/sci/servlets/ServletContainerInitializerImpl.java]: Clean up whatever git is complaining about here

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat,io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.6.0_fat_toleration.lWAS,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Review by pnicolucci — APPROVED** [review]

**Comment by pmd1nh** [comment]
one failed but unrelated test

Name Tests Failures Errors Time Taken Run Order
					
no component 94266 0 1
	
com.ibm.ws.ui.tool.serverConfig_fat.gui.nd.rw FAT tests 17 0 1
	
com.ibm.ws.ui.tool.serverConfig.fat.gui.nd.rw.tests.ServerConfigValidatorTest 14 0 1 11 m 05 s 2

validationUsingDefaultAuthentication 1 0 1 39 s 2

**Comment by pmd1nh** [comment]
#libby

**Pull request merged**